src/kademlia/get_item.cpp

```diff
@@ -70,6 +70,15 @@ void get_item::got_data(bdecode_node const& v,
 		{
 			if (!m_data.assign(v, salt, seq, pk, sig))
 				return;
+
+			if (!m_put_requested)
+			{
+				// For get_item, we should call callback when we get data.
+				// The data maybe is not the latest, but we can update it later.
+				// Waitting transaction timeout (15 seconds) is not efficient.
+				// For put_item, only call the callback at the end.
+				m_data_callback(m_data);
```

Comment:
I would have thought the callback would have to be told whether the data is authoritative too. For any non-authoritative data we cannot perform a put for instance. The callback is supposed to modify the data and return true if it wants to put. It's possible that it finds the data in the state it wants it to be, in which case it may not want to "put".